x-pack/plugins/apm/server/lib/index_pattern/index.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Server } from 'hapi';
+// @ts-ignore
+import { getSavedObjects } from '../../../../../../src/legacy/core_plugins/kibana/server/tutorials/apm/saved_objects/get_saved_objects';
+import { getSavedObjectsClient } from '../saved_objects/client';
+
+export async function ensureIndexPatternExists(server: Server) {
+  const config = server.config();
+  const apmIndexPatternTitle = config.get('apm_oss.indexPattern');
+  const savedObjectsClient = getSavedObjectsClient(server);
+  const apmIndexPatternSearchResp = await savedObjectsClient.find({
+    type: 'index-pattern',
+    search: `"${apmIndexPatternTitle}"`,
+    searchFields: ['title'],
+    perPage: 200
+  });
+  if (apmIndexPatternSearchResp.total === 0) {
+    const savedObjects = getSavedObjects(apmIndexPatternTitle);
+    await savedObjectsClient.bulkCreate(savedObjects, { overwrite: false });
```

Comment:
I'm wondering if we should just ditch the logic above, and create the index pattern every time - if we keep `overwrite: false` it should be the same, right?

Reply:
Btw. i can see that POSTing a big JSON file is heavier than doing a light GET. But since we only do it on startup I don't think it's that bad.